src/window.rs

```diff
@@ -0,0 +1,162 @@
+//! Displays an image in a window created by sdl2.
+
+use image::RgbaImage;
+use sdl2::event::{Event, WindowEvent};
+use sdl2::keyboard::Keycode;
+use sdl2::pixels::{Color, PixelFormatEnum};
+use sdl2::rect::Rect;
+use sdl2::surface::Surface;
+
+/// Displays the provided RGBA image in a new window.
+/// Minimum window size is 150 x 150.
+pub fn display_image(title: &str, image: &RgbaImage, window_width: u32, window_height: u32) {
+    const MIN_WINDOW_DIMENSION: u32 = 150;
+    // ensures window size is minimum size, so that image resizing calculations for the window are correct
+    let window_width: u32 = if window_width < MIN_WINDOW_DIMENSION {
```

Comment:
u32 implements Ord, so this can just be `let window_width = window_width.max(MIN_WINDOW_DIMENSION)`